### PR TITLE
Avoid possible leaks in `readConfig()`

### PR DESCRIPTION
**Pull request by dtardon**
Follow-up for #126.

### CODE REVIEW

File: alternatives.c [hunk 1]
```diff
@@ -485,7 +485,7 @@ static int readConfig(struct alternativeSet *set, const char *title,
             newAlt.followers[i - 1].target = (line && strlen(line)) ? strsteal(&line) : NULL;
         }
 
-        if (newAlt.priority > set->alts[set->best].priority)
+        if (set->best == -1 || newAlt.priority > set->alts[set->best].priority)
```
Comment (dtardon) [inline alternatives.c:488]: This could also be moved a bit down beyond line 493. Then it would continue to work with the initial `0`.

Comment (lnykryn) [reply alternatives.c:488]: Yep, you are right, feel free to force-push a removal of this one and amend your old one.

Comment (dtardon) [reply alternatives.c]: Done.

File: alternatives.c [hunk 2]
```diff
@@ -486,11 +486,13 @@ static int readConfig(struct alternativeSet *set, const char *title,
         }
 
         set->alts = realloc(set->alts, (set->numAlts + 1) * sizeof(*set->alts));
-        set->alts[set->numAlts++] = newAlt;
+        set->alts[set->numAlts] = newAlt;
 
-        if (newAlt.priority > set->alts[set->best].priority)
+        if (set->best == -1 || newAlt.priority > set->alts[set->best].priority)
             set->best = set->numAlts;
```
Comment (dtardon) [inline alternatives.c:492]: Ah, I see... I missed that off-by-one. Fixed in the 2nd commit, where it really belongs.

**Review by lnykryn — APPROVED** [review]

**Pull request merged**